{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f098cd00",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "183b17ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "METADATA_FILE_PATH ='../data/E-MTAB-3321/raw/E-MTAB-3321.sdrf.txt'\n",
    "COUNTS_FILE_PATH = '../data/E-MTAB-3321/raw/Goolam_et_al_2015_count_table.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f3fb64e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_metadata = pd.read_csv(METADATA_FILE_PATH, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c20c758a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Source Name', 'Comment[ENA_SAMPLE]', 'Characteristics[organism]',\n",
       "       'Characteristics[strain]', 'Characteristics[developmental stage]',\n",
       "       'Characteristics[organism part]', 'Characteristics[cell type]',\n",
       "       'Characteristics[division pattern]',\n",
       "       'Characteristics[single cell well quality]', 'Characteristics[batch]',\n",
       "       'Material Type', 'Protocol REF', 'Protocol REF.1', 'Protocol REF.2',\n",
       "       'Extract Name', 'Material Type.1', 'Comment[LIBRARY_LAYOUT]',\n",
       "       'Comment[LIBRARY_SELECTION]', 'Comment[LIBRARY_SOURCE]',\n",
       "       'Comment[LIBRARY_STRAND]', 'Comment[LIBRARY_STRATEGY]',\n",
       "       'Comment[NOMINAL_LENGTH]', 'Comment[NOMINAL_SDEV]',\n",
       "       'Comment[single cell isolation]', 'Comment[library construction]',\n",
       "       'Comment[end bias]', 'Comment[input molecule]', 'Comment[primer]',\n",
       "       'Comment[spike in]', 'Comment[spike in dilution]',\n",
       "       'Comment[spike sequence file]', 'Comment[spike concentration file]',\n",
       "       'Protocol REF.3', 'Performer', 'Assay Name', 'Technology Type',\n",
       "       'Comment[ENA_EXPERIMENT]', 'Scan Name', 'Comment[SUBMITTED_FILE_NAME]',\n",
       "       'Comment[ENA_RUN]', 'Comment[FASTQ_URI]', 'Comment[SPOT_LENGTH]',\n",
       "       'Comment[READ_INDEX_1_BASE_COORD]', 'Derived Array Data File',\n",
       "       'Comment [Derived ArrayExpress FTP file]',\n",
       "       'Factor Value[developmental stage]', 'Factor Value[division pattern]',\n",
       "       'Factor Value[single cell identifier]'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b886a194",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source Name</th>\n",
       "      <th>Comment[ENA_SAMPLE]</th>\n",
       "      <th>Characteristics[organism]</th>\n",
       "      <th>Characteristics[strain]</th>\n",
       "      <th>Characteristics[developmental stage]</th>\n",
       "      <th>Characteristics[organism part]</th>\n",
       "      <th>Characteristics[cell type]</th>\n",
       "      <th>Characteristics[division pattern]</th>\n",
       "      <th>Characteristics[single cell well quality]</th>\n",
       "      <th>Characteristics[batch]</th>\n",
       "      <th>...</th>\n",
       "      <th>Comment[SUBMITTED_FILE_NAME]</th>\n",
       "      <th>Comment[ENA_RUN]</th>\n",
       "      <th>Comment[FASTQ_URI]</th>\n",
       "      <th>Comment[SPOT_LENGTH]</th>\n",
       "      <th>Comment[READ_INDEX_1_BASE_COORD]</th>\n",
       "      <th>Derived Array Data File</th>\n",
       "      <th>Comment [Derived ArrayExpress FTP file]</th>\n",
       "      <th>Factor Value[developmental stage]</th>\n",
       "      <th>Factor Value[division pattern]</th>\n",
       "      <th>Factor Value[single cell identifier]</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2cell_1_A</td>\n",
       "      <td>ERS1043823</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6 x CBA</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>embryo</td>\n",
       "      <td>blastoderm cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OK</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Goolam_et_al_2015_Batch1_Sample1_1.fq.gz</td>\n",
       "      <td>ERR1229132</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/002/...</td>\n",
       "      <td>200</td>\n",
       "      <td>101</td>\n",
       "      <td>Goolam_et_al_2015_count_table.tsv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>2cell_1_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2cell_1_A</td>\n",
       "      <td>ERS1043823</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6 x CBA</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>embryo</td>\n",
       "      <td>blastoderm cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OK</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Goolam_et_al_2015_Batch1_Sample1_2.fq.gz</td>\n",
       "      <td>ERR1229132</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/002/...</td>\n",
       "      <td>200</td>\n",
       "      <td>101</td>\n",
       "      <td>Goolam_et_al_2015_count_table.tsv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>2cell_1_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2cell_1_B</td>\n",
       "      <td>ERS1043824</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6 x CBA</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>embryo</td>\n",
       "      <td>blastoderm cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OK</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Goolam_et_al_2015_Batch1_Sample13_1.fq.gz</td>\n",
       "      <td>ERR1229133</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/003/...</td>\n",
       "      <td>200</td>\n",
       "      <td>101</td>\n",
       "      <td>Goolam_et_al_2015_count_table.tsv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>2cell_1_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2cell_1_B</td>\n",
       "      <td>ERS1043824</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6 x CBA</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>embryo</td>\n",
       "      <td>blastoderm cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OK</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Goolam_et_al_2015_Batch1_Sample13_2.fq.gz</td>\n",
       "      <td>ERR1229133</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/003/...</td>\n",
       "      <td>200</td>\n",
       "      <td>101</td>\n",
       "      <td>Goolam_et_al_2015_count_table.tsv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>2cell_1_B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2cell_2_A</td>\n",
       "      <td>ERS1043825</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>C57BL/6 x CBA</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>embryo</td>\n",
       "      <td>blastoderm cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>OK</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>Goolam_et_al_2015_Batch1_Sample25_1.fq.gz</td>\n",
       "      <td>ERR1229134</td>\n",
       "      <td>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/004/...</td>\n",
       "      <td>200</td>\n",
       "      <td>101</td>\n",
       "      <td>Goolam_et_al_2015_count_table.tsv</td>\n",
       "      <td>ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...</td>\n",
       "      <td>cleavage 2-cell</td>\n",
       "      <td>not applicable</td>\n",
       "      <td>2cell_2_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Name Comment[ENA_SAMPLE] Characteristics[organism]  \\\n",
       "0   2cell_1_A          ERS1043823              Mus musculus   \n",
       "1   2cell_1_A          ERS1043823              Mus musculus   \n",
       "2   2cell_1_B          ERS1043824              Mus musculus   \n",
       "3   2cell_1_B          ERS1043824              Mus musculus   \n",
       "4   2cell_2_A          ERS1043825              Mus musculus   \n",
       "\n",
       "  Characteristics[strain] Characteristics[developmental stage]  \\\n",
       "0           C57BL/6 x CBA                      cleavage 2-cell   \n",
       "1           C57BL/6 x CBA                      cleavage 2-cell   \n",
       "2           C57BL/6 x CBA                      cleavage 2-cell   \n",
       "3           C57BL/6 x CBA                      cleavage 2-cell   \n",
       "4           C57BL/6 x CBA                      cleavage 2-cell   \n",
       "\n",
       "  Characteristics[organism part] Characteristics[cell type]  \\\n",
       "0                         embryo            blastoderm cell   \n",
       "1                         embryo            blastoderm cell   \n",
       "2                         embryo            blastoderm cell   \n",
       "3                         embryo            blastoderm cell   \n",
       "4                         embryo            blastoderm cell   \n",
       "\n",
       "  Characteristics[division pattern] Characteristics[single cell well quality]  \\\n",
       "0                    not applicable                                        OK   \n",
       "1                    not applicable                                        OK   \n",
       "2                    not applicable                                        OK   \n",
       "3                    not applicable                                        OK   \n",
       "4                    not applicable                                        OK   \n",
       "\n",
       "   Characteristics[batch]  ...               Comment[SUBMITTED_FILE_NAME]  \\\n",
       "0                       1  ...   Goolam_et_al_2015_Batch1_Sample1_1.fq.gz   \n",
       "1                       1  ...   Goolam_et_al_2015_Batch1_Sample1_2.fq.gz   \n",
       "2                       1  ...  Goolam_et_al_2015_Batch1_Sample13_1.fq.gz   \n",
       "3                       1  ...  Goolam_et_al_2015_Batch1_Sample13_2.fq.gz   \n",
       "4                       1  ...  Goolam_et_al_2015_Batch1_Sample25_1.fq.gz   \n",
       "\n",
       "  Comment[ENA_RUN]                                 Comment[FASTQ_URI]  \\\n",
       "0       ERR1229132  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/002/...   \n",
       "1       ERR1229132  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/002/...   \n",
       "2       ERR1229133  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/003/...   \n",
       "3       ERR1229133  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/003/...   \n",
       "4       ERR1229134  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR122/004/...   \n",
       "\n",
       "  Comment[SPOT_LENGTH] Comment[READ_INDEX_1_BASE_COORD]  \\\n",
       "0                  200                              101   \n",
       "1                  200                              101   \n",
       "2                  200                              101   \n",
       "3                  200                              101   \n",
       "4                  200                              101   \n",
       "\n",
       "             Derived Array Data File  \\\n",
       "0  Goolam_et_al_2015_count_table.tsv   \n",
       "1  Goolam_et_al_2015_count_table.tsv   \n",
       "2  Goolam_et_al_2015_count_table.tsv   \n",
       "3  Goolam_et_al_2015_count_table.tsv   \n",
       "4  Goolam_et_al_2015_count_table.tsv   \n",
       "\n",
       "             Comment [Derived ArrayExpress FTP file]  \\\n",
       "0  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   \n",
       "1  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   \n",
       "2  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   \n",
       "3  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   \n",
       "4  ftp://ftp.ebi.ac.uk/pub/databases/microarray/d...   \n",
       "\n",
       "  Factor Value[developmental stage] Factor Value[division pattern]  \\\n",
       "0                   cleavage 2-cell                 not applicable   \n",
       "1                   cleavage 2-cell                 not applicable   \n",
       "2                   cleavage 2-cell                 not applicable   \n",
       "3                   cleavage 2-cell                 not applicable   \n",
       "4                   cleavage 2-cell                 not applicable   \n",
       "\n",
       "  Factor Value[single cell identifier]  \n",
       "0                            2cell_1_A  \n",
       "1                            2cell_1_A  \n",
       "2                            2cell_1_B  \n",
       "3                            2cell_1_B  \n",
       "4                            2cell_2_A  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e6432b4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Characteristics[developmental stage]\n",
       "cleavage 4-cell     128\n",
       "cleavage 8-cell      64\n",
       "cleavage 2-cell      32\n",
       "cleavage 16-cell     12\n",
       "cleavage 32-cell     12\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_metadata['Characteristics[developmental stage]'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "07b11be2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Characteristics[cell type]\n",
       "blastoderm cell    248\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_metadata['Characteristics[cell type]'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "75ca65c0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Factor Value[single cell identifier]\n",
       "2cell_1_A       2\n",
       "ME_4cell_6_C    2\n",
       "8cell_2_D       2\n",
       "8cell_2_C       2\n",
       "8cell_2_B       2\n",
       "               ..\n",
       "ME_4cell_2_A    2\n",
       "ME_4cell_1_D    2\n",
       "ME_4cell_1_C    2\n",
       "ME_4cell_1_B    2\n",
       "32cell_F        2\n",
       "Name: count, Length: 124, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_metadata['Factor Value[single cell identifier]'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2077c7a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2cell_1_A</th>\n",
       "      <th>2cell_1_B</th>\n",
       "      <th>2cell_2_A</th>\n",
       "      <th>2cell_2_B</th>\n",
       "      <th>2cell_3_A</th>\n",
       "      <th>2cell_3_B</th>\n",
       "      <th>ME_4cell_1_A</th>\n",
       "      <th>ME_4cell_1_B</th>\n",
       "      <th>ME_4cell_1_C</th>\n",
       "      <th>ME_4cell_1_D</th>\n",
       "      <th>...</th>\n",
       "      <th>EM_4cell_5_C</th>\n",
       "      <th>EM_4cell_5_D</th>\n",
       "      <th>EM_4cell_6_A</th>\n",
       "      <th>EM_4cell_6_B</th>\n",
       "      <th>EM_4cell_6_C</th>\n",
       "      <th>EM_4cell_6_D</th>\n",
       "      <th>ME_4cell_6_A</th>\n",
       "      <th>ME_4cell_6_B</th>\n",
       "      <th>ME_4cell_6_C</th>\n",
       "      <th>ME_4cell_6_D</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000001</th>\n",
       "      <td>165</td>\n",
       "      <td>79</td>\n",
       "      <td>124</td>\n",
       "      <td>41</td>\n",
       "      <td>107</td>\n",
       "      <td>167</td>\n",
       "      <td>84</td>\n",
       "      <td>70</td>\n",
       "      <td>28</td>\n",
       "      <td>42</td>\n",
       "      <td>...</td>\n",
       "      <td>63</td>\n",
       "      <td>98</td>\n",
       "      <td>25</td>\n",
       "      <td>21</td>\n",
       "      <td>58</td>\n",
       "      <td>63</td>\n",
       "      <td>109</td>\n",
       "      <td>82</td>\n",
       "      <td>33</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000028</th>\n",
       "      <td>722</td>\n",
       "      <td>547</td>\n",
       "      <td>481</td>\n",
       "      <td>219</td>\n",
       "      <td>808</td>\n",
       "      <td>483</td>\n",
       "      <td>221</td>\n",
       "      <td>280</td>\n",
       "      <td>158</td>\n",
       "      <td>222</td>\n",
       "      <td>...</td>\n",
       "      <td>164</td>\n",
       "      <td>223</td>\n",
       "      <td>188</td>\n",
       "      <td>139</td>\n",
       "      <td>242</td>\n",
       "      <td>162</td>\n",
       "      <td>186</td>\n",
       "      <td>284</td>\n",
       "      <td>409</td>\n",
       "      <td>218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000031</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000000037</th>\n",
       "      <td>58</td>\n",
       "      <td>33</td>\n",
       "      <td>22</td>\n",
       "      <td>34</td>\n",
       "      <td>68</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC-00164</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC-00165</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC-00168</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC-00170</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERCC-00171</th>\n",
       "      <td>23</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "      <td>36</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "      <td>37</td>\n",
       "      <td>...</td>\n",
       "      <td>254</td>\n",
       "      <td>601</td>\n",
       "      <td>268</td>\n",
       "      <td>228</td>\n",
       "      <td>257</td>\n",
       "      <td>281</td>\n",
       "      <td>406</td>\n",
       "      <td>573</td>\n",
       "      <td>428</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41480 rows × 124 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    2cell_1_A  2cell_1_B  2cell_2_A  2cell_2_B  2cell_3_A  \\\n",
       "ENSMUSG00000000001        165         79        124         41        107   \n",
       "ENSMUSG00000000003          0          0          0          0          0   \n",
       "ENSMUSG00000000028        722        547        481        219        808   \n",
       "ENSMUSG00000000031          0          0          0          0          0   \n",
       "ENSMUSG00000000037         58         33         22         34         68   \n",
       "...                       ...        ...        ...        ...        ...   \n",
       "ERCC-00164                  0          0          0          0          0   \n",
       "ERCC-00165                  0          0          0          1          0   \n",
       "ERCC-00168                  0          0          0          0          0   \n",
       "ERCC-00170                  0          0          0          0          0   \n",
       "ERCC-00171                 23         21         11         17         36   \n",
       "\n",
       "                    2cell_3_B  ME_4cell_1_A  ME_4cell_1_B  ME_4cell_1_C  \\\n",
       "ENSMUSG00000000001        167            84            70            28   \n",
       "ENSMUSG00000000003          0             0             0             0   \n",
       "ENSMUSG00000000028        483           221           280           158   \n",
       "ENSMUSG00000000031          0             0             0             0   \n",
       "ENSMUSG00000000037         27             0             9             5   \n",
       "...                       ...           ...           ...           ...   \n",
       "ERCC-00164                  0             0             0             0   \n",
       "ERCC-00165                  0             4             0             0   \n",
       "ERCC-00168                  0             0             0             0   \n",
       "ERCC-00170                  0             2             0             0   \n",
       "ERCC-00171                 18            24            11            17   \n",
       "\n",
       "                    ME_4cell_1_D  ...  EM_4cell_5_C  EM_4cell_5_D  \\\n",
       "ENSMUSG00000000001            42  ...            63            98   \n",
       "ENSMUSG00000000003             0  ...             0             0   \n",
       "ENSMUSG00000000028           222  ...           164           223   \n",
       "ENSMUSG00000000031             0  ...             0             0   \n",
       "ENSMUSG00000000037            10  ...            10             8   \n",
       "...                          ...  ...           ...           ...   \n",
       "ERCC-00164                     0  ...             0             0   \n",
       "ERCC-00165                     3  ...            18            38   \n",
       "ERCC-00168                     0  ...             0             0   \n",
       "ERCC-00170                     0  ...             0             0   \n",
       "ERCC-00171                    37  ...           254           601   \n",
       "\n",
       "                    EM_4cell_6_A  EM_4cell_6_B  EM_4cell_6_C  EM_4cell_6_D  \\\n",
       "ENSMUSG00000000001            25            21            58            63   \n",
       "ENSMUSG00000000003             0             0             0             0   \n",
       "ENSMUSG00000000028           188           139           242           162   \n",
       "ENSMUSG00000000031             0             0             0             0   \n",
       "ENSMUSG00000000037             9             0             5             0   \n",
       "...                          ...           ...           ...           ...   \n",
       "ERCC-00164                     0             0             0             0   \n",
       "ERCC-00165                     0             6             3             7   \n",
       "ERCC-00168                     1             0             0             0   \n",
       "ERCC-00170                     0             0             6            28   \n",
       "ERCC-00171                   268           228           257           281   \n",
       "\n",
       "                    ME_4cell_6_A  ME_4cell_6_B  ME_4cell_6_C  ME_4cell_6_D  \n",
       "ENSMUSG00000000001           109            82            33            43  \n",
       "ENSMUSG00000000003             0             0             0             0  \n",
       "ENSMUSG00000000028           186           284           409           218  \n",
       "ENSMUSG00000000031             0             0             0             0  \n",
       "ENSMUSG00000000037            12            13             0             6  \n",
       "...                          ...           ...           ...           ...  \n",
       "ERCC-00164                     0             0             0             0  \n",
       "ERCC-00165                     7             0             0            20  \n",
       "ERCC-00168                     0             0             0             0  \n",
       "ERCC-00170                     0             8             2             1  \n",
       "ERCC-00171                   406           573           428           327  \n",
       "\n",
       "[41480 rows x 124 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_df = pd.read_csv(COUNTS_FILE_PATH, sep='\\t')\n",
    "counts_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1624e9b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "cells_by_genes = counts_df.T\n",
    "cells_by_genes.to_csv('../data/E-MTAB-3321/raw/E-MTAB-3321.csv.gz', compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b5118a3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "scrna-seq-clustering",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
